### PR TITLE
Add cql support for confluence connector

### PR DESCRIPTION
**Pull request by hagen-danswer**

### CODE REVIEW

File: backend/danswer/connectors/confluence/connector.py [hunk 1]
```diff
@@ -56,8 +57,55 @@
 )
 
 
+class DanswerConfluence(Confluence):
```
Comment (hagen-danswer) [inline backend/danswer/connectors/confluence/connector.py:60]: this class is necessary to make cql quries work. read comment

File: backend/danswer/connectors/confluence/connector.py [hunk 2]
```diff
@@ -56,8 +57,103 @@
...
+            response = self.get(url_suffix)
+            return response.get("results", [])
+        except Exception as e:
+            raise e
+
+
+def _replace_cql_time_filter(
+    cql_query: str, start_time: datetime, end_time: datetime
+) -> str:
+    """
```
Comment (hagen-danswer) [inline backend/danswer/connectors/confluence/connector.py:94]: this ensures that the cql properly contains no overlapping lastmodified query parameters

File: web/src/lib/connectors/connectors.tsx [hunk 3]
```diff
@@ -223,22 +245,21 @@ export const connectorConfigs: Record<
...
+    advanced_values: [],
   },
   confluence: {
     description: "Configure Confluence connector",
-    subtext: `Specify the base URL of your Confluence instance, the space name, and optionally a specific page ID to index. If no page ID is provided, the entire space will be indexed.
-
-For example, entering "https://your-company.atlassian.net/wiki" as the Wiki Base URL, "KB" as the Space, and "164331" as the Page ID will index the specific page at https:///your-company.atlassian.net/wiki/spaces/KB/pages/164331/Page. If you leave the Page ID empty, it will index the entire KB space.
-
-Selecting the "Index Recursively" checkbox will index the specified page and all of its children.`,
+    subtext: `Specify the base URL of your Confluence instance, the space name, and optionally a specific page ID to index. If no page ID is provided, the entire space will be indexed. If no space is specified, all available Confluence spaces will be indexed.`,
```
Comment (hagen-danswer) [inline web/src/lib/connectors/connectors.tsx:262]: less yapping

File: web/src/app/admin/connectors/[connector]/pages/DynamicConnectorCreationForm.tsx [hunk 4]
```diff
@@ -23,6 +24,61 @@ const DynamicConnectionForm: FC<DynamicConnectionFormProps> = ({
   setSelectedFiles,
   values,
 }) => {
```
Comment (hagen-danswer) [inline web/src/app/admin/connectors/[connector]/pages/DynamicConnectorCreationForm.tsx:26]: added an advanced_values section

**Review by Weves — APPROVED** [review]

**Pull request merged**